### ISSUE TITLE
End User auth using Auth0 and GitKit

### ISSUE THREAD

**Issue by jasminejaksic** [istio/auth#43]
Tracking issue

**Comment by mwitkow** [comment]
Hi,

Congrats on the project, and I wanted to ask a follow up question as to how you want to structure End User Auth. We have a way of issuing developer (and operator) certifiacets using [Vault](https://www.vaultproject.io/api/secret/pki/index.html)'s dynamic PKI mechanism. This, combined with an edge mutual-TLS proxy for our Kubernetes clusters allows us to grant access to, for example, debug endpoints of our services (a `/statusz` or the Golang `/debug/request`).

I was wondering whether the auth manager is considering using client TLS certs for End User auth, maybe through some SPIFEE naming convention. For example:
`spiffee://internal.improbable.io/users/michal`

We thought about using OIDC identity tokens  similarly to [Kubernetes OIDC auth](https://kubernetes.io/docs/admin/authentication/#option-1---oidc-authenticator) but it has practical drawbacks: you need the proxy to do a complicated OAuth2 flow for browsers (for human operators) to be able to use it. Alternatively, a `Proxy-Authentication` header can be used, but that kinda defeats the purpose of mutual TLS.

What are your thoughts on doing this in Istio?

**Comment by myidpt** [comment]
That's a very good question.

For the End User Auth, we planned to support OIDC identity tokens, so the auth will be JWT+TLS. I agree it has drawbacks, but it is much easier for non-developer users to use (no need to get certificate through CSR to CA). For developers, we may support certs with user-accounts as identities (but it is still on discussion).

**Comment by myidpt** [comment]
Also see this for how we want to support debugging in the short term:
https://github.com/istio/istio/issues/242

**Comment by wattli** [comment]
@lizan , are you working on this?

**Comment by lizan** [comment]
Yes, I'm proposing a Workgroup for this:
https://groups.google.com/forum/#!topic/istio-security/pK0oUikOZuw

**Comment by kdillane** [comment]
Does end user auth also pertain to external systems accessing a Kubernetes cluster through the Ingress?  Being able to apply policy decisions about callers into a cluster isn't obvious from an implementation perspective and seems pertinent to both end-users and non-cluster external clients.

**Comment by kdillane** [comment]
Are there any updates in this area?  I know there has been a lot of movement in the Istio development space and I want to make sure this is still a priority.  Thanks!

**Comment by lizan** [comment]
Yes this is still on the track. It is currently in a pre-alpha stage along with istio auth policy. Feel free to join security SIG meeting to get development updates.
